### ISSUE TITLE
Type check super() arguments

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#526]
Currently the arguments in `super()` calls are not checked.

**Comment by JukkaL** [comment]
For example, mypy doesn't report an error for this code:

```
class A:
    def f(self) -> None: pass

class B(A):
    def f(self) -> None:
        super(1, '', '').f()   # ouch, should generate an error
```

**Comment by axitkhurana** [comment]
Looking into this one

**Comment by JukkaL** [comment]
This would actually be pretty nice to have as Python 2 support is going strong.

**Comment by rwbarton** [comment]
I remember @axitkhurana and I came up with a nice plan for this at PyCon. @axitkhurana did you end up getting it working? I can also reconstruct it if needed.
